File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/implement/SizeConvertImp.java
package com.platdmit.simplecloudmanager.domain.converters.implement;

import com.platdmit.simplecloudmanager.data.api.models.ApiSize;
import com.platdmit.simplecloudmanager.data.database.entity.DbSize;
import com.platdmit.simplecloudmanager.domain.converters.SizeConverter;
import com.platdmit.simplecloudmanager.domain.models.Size;

import java.util.ArrayList;
import java.util.List;

public class SizeConvertImp implements SizeConverter {
    @Override
    public DbSize fromApiToDb(ApiSize apiSize) {
        return new DbSize(
                apiSize.getId(),
                apiSize.getSlug(),
                apiSize.getMemory(),
                apiSize.getVcpus(),
                apiSize.getDisk(),
                apiSize.getTransfer(),
                apiSize.getPriceMonthly(),
                apiSize.getPriceHourly(),
                apiSize.getLinked(),
                apiSize.getMain(),
                apiSize.isTest(),
                apiSize.isArchive()
        );
    }

    @Override
    public Size fromDbToDomain(DbSize dbSize) {
        return new Size(
                dbSize.getId(),
                dbSize.getNetId(),
                dbSize.getSlug(),
                dbSize.getMemory(),
                dbSize.getVcpus(),
                dbSize.getDisk(),
                dbSize.getTransfer(),
                dbSize.getPriceMonthly(),
                dbSize.getPriceHourly(),
                dbSize.getLinked(),
                dbSize.getMain(),
                dbSize.isTest(),
                dbSize.isArchive()
        );
    }

    @Override
    public List<Size> fromDbToDomainList(List<DbSize> dbList) {
        List<Size> convertList = new ArrayList<>();
        for (DbSize dbSize : dbList) {
            convertList.add(fromDbToDomain(dbSize));
        }
        return convertList;
    }

    @Override
    public List<DbSize> fromApiToDbList(List<ApiSize> apiList) {
        List<DbSize> convertList = new ArrayList<>();
        for (ApiSize apiSize : apiList) {
            convertList.add(fromApiToDb(apiSize));
        }
        return convertList;
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/domain/converters/implement/SizeConvertImp.kt
package com.platdmit.simplecloudmanager.domain.converters.implement

import com.platdmit.data.api.models.ApiSize
import com.platdmit.data.database.entity.DbSize
import com.platdmit.simplecloudmanager.domain.converters.SizeConverter
import com.platdmit.simplecloudmanager.domain.models.Size

class SizeConvertImp : SizeConverter {
    override fun fromApiToDb(apiSize: ApiSize): DbSize {
        return DbSize(
                apiSize.id.toLong(),
                apiSize.slug,
                apiSize.memory,
                apiSize.vcpus,
                apiSize.disk,
                apiSize.transfer,
                apiSize.priceMonthly,
                apiSize.priceHourly,
                apiSize.linked,
                apiSize.main,
                apiSize.isTest,
                apiSize.isArchive
        )
    }

    override fun fromDbToDomain(dbSize: DbSize): Size {
        return Size(
                dbSize.id,
                dbSize.id.toString(),
                dbSize.slug,
                dbSize.memory,
                dbSize.vcpus,
                dbSize.disk,
                dbSize.transfer,
                dbSize.priceMonthly,
                dbSize.priceHourly,
                dbSize.linked,
                dbSize.main,
                dbSize.isTest,
                dbSize.isArchive
        )
    }

    override fun fromDbToDomainList(dbList: List<DbSize>): List<Size> = dbList.map { fromDbToDomain(it) }

    override fun fromApiToDbList(apiList: List<ApiSize>): List<DbSize> = apiList.map { fromApiToDb(it) }
}